client/dashboard/data/site-logs.ts

```diff
@@ -0,0 +1,104 @@
+import wpcom from 'calypso/lib/wp';
+
+export interface SiteLogsAPIResponse {
+	message: string;
+	data: {
+		total_results: number | { value: number; relation: string };
+		logs: ( PHPLogFromEndpoint | ServerLogFromEndpoint )[];
+	};
+}
+
+export interface SiteLogsData {
+	total_results: number;
+	logs: ( ServerLog | PHPLog )[];
+	has_more: boolean;
+}
+
+export interface PHPLog extends Omit< PHPLogFromEndpoint, 'atomic_site_id' > {
+	id: string;
+}
+
+export enum LogType {
+	PHP = 'php',
+	SERVER = 'server',
+}
+
+export interface LogQueryParams {
+	from: string;
+	to: string;
+	// PHP filters
+	severity: string;
+	// Server filters
+	request_type: string;
+	status: string;
+	renderer: string;
+	cached: string;
+}
+
+export interface FilterType {
+	[ key: string ]: Array< string >;
+}
+
+export interface ServerLogFromEndpoint {
+	date: string;
+	request_type: 'GET' | 'HEAD' | 'POST' | 'PUT' | 'DELETE';
+	status: '200' | '301' | '302' | '400' | '401' | '403' | '404' | '429' | '500';
+	request_url: string;
+	body_bytes_sent: number;
+	cached: string;
+	http_host: string;
+	http_referer: string;
+	http2: string;
+	http_user_agent: string;
+	http_version: string;
+	http_x_forwarded_for: string;
+	renderer: string;
+	request_completion: string;
+	request_time: string;
+	scheme: string;
+	timestamp: number;
+	type: string;
+	user_ip: string;
+}
+
+export interface ServerLog extends ServerLogFromEndpoint {
+	id: string;
```

Comment:
I wonder why this `id` is for... can we just utilize the `timestamp` field for that?

Reply:
When mapping in `siteLogsQuery` the returned data now matches the `ServerLog` / `PHPLog` types due to the added id (same as what is was before).
It seems like the timestamp itself wouldn't necessarily be unique 🤔

Reply:
I see... we ideally should add this in the backend side, but we can do it later. 😄

Reply:
Good point I added a task for it at DOTCOM-14208